examples/lightgbm_link_baseline.py

```diff
@@ -0,0 +1,315 @@
+import argparse
+from collections import Counter
+from typing import Dict
+
+import numpy as np
+import pandas as pd
+import torch
+import torch_frame
+from inferred_stypes import dataset2inferred_stypes
+from text_embedder import GloveTextEmbedding
+from torch_frame.config.text_embedder import TextEmbedderConfig
+from torch_frame.data import Dataset
+from torch_frame.gbdt import LightGBM
+from torch_frame.typing import Metric
+
+from relbench.data import LinkTask, RelBenchDataset, RelBenchLinkTask, Table
+from relbench.datasets import get_dataset
+
+LINK_PRED_BASELINE_TARGET_COL_NAME = "link_pred_baseline_target_column_name"
+PRED_SCORE_COL_NAME = "pred_score_col_name"
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--dataset", type=str, default="rel-stackex")
+parser.add_argument("--task", type=str, default="rel-stackex-comment-on-post")
+
+args = parser.parse_args()
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+dataset: RelBenchDataset = get_dataset(name=args.dataset, process=True)
+task: RelBenchLinkTask = dataset.get_task(args.task, process=True)
+target_col_name: str = LINK_PRED_BASELINE_TARGET_COL_NAME
+
+train_table = task.train_table
+val_table = task.val_table
+test_table = task.test_table
+
+# We plan to merge train table with entity and target table to include both
+# entity and target table features during lightGBM training.
+dfs: Dict[str, pd.DataFrame] = {}
+target_dfs: Dict[str, pd.DateOffset] = {}
+lhs_entity_table = dataset.db.table_dict[task.src_entity_table]
+lhs_entity_df = lhs_entity_table.df
+rhs_entity_table = dataset.db.table_dict[task.dst_entity_table]
+rhs_entity_df = rhs_entity_table.df
+
+# Prepare col_to_stype dictioanry mapping between column names and stypes
+# for torch_frame Dataset initialization.
+col_to_stype = {}
+lhs_entity_table_col_to_stype = dataset2inferred_stypes[args.dataset][
+    task.src_entity_table
+]
+rhs_entity_table_col_to_stype = dataset2inferred_stypes[args.dataset][
+    task.dst_entity_table
+]
+
+if lhs_entity_table.pkey_col is not None:
+    del lhs_entity_table_col_to_stype[lhs_entity_table.pkey_col]
+for fkey_col in lhs_entity_table.fkey_col_to_pkey_table.keys():
+    del lhs_entity_table_col_to_stype[fkey_col]
+if rhs_entity_table.pkey_col is not None:
+    del rhs_entity_table_col_to_stype[rhs_entity_table.pkey_col]
+for fkey_col in rhs_entity_table.fkey_col_to_pkey_table.keys():
+    del rhs_entity_table_col_to_stype[fkey_col]
```

Comment:
There is a similar logic [here](https://github.com/snap-stanford/relbench/blob/main/relbench/external/graph.py#L79-L86).

Can we create some util function [here](https://github.com/snap-stanford/relbench/blob/main/relbench/external/utils.py)? something like 
```python
def remove_pkey_fkey(col_to_stype, table) -> dict:
   # return col_to_stype where pkey and fkey are removed.
```